geoips/pydantic/sectors.py

```diff
@@ -0,0 +1,601 @@
+            le=90,
+            description="Upper right latitude in degrees.",
+        ),
+    ]
+    max_lon: Annotated[
+        float,
+        Field(
+            ...,
+            strict=True,
+            ge=-180,
+            le=180,
+            description="Upper right longitude in degrees.",
+        ),
+    ]
+
+    box_resolution_km: Annotated[
+        float,
+        Field(
+            ...,
+            strict=True,
+            gt=0,
+            description="The resolution of each pixel in kilometers.",
+        ),
+    ]
+
+
+class StitchedMetadata(StaticMetadata):
+    """Metadata for stitched imagery sectors."""
+
+    primary_area_definition: str = Field(
+        ..., description="Name of the area definition to be used."
+    )
+
+
+class TCMetadata(FrozenModel):
+    """Metdata format for Tropical Cyclone sectors."""
+
+    pressure: Annotated[
+        float,
+        Field(
+            ..., strict=True, gt=0, description="Pressure of the storm in millibars."
+        ),
+    ]
+    vmax: Annotated[
+        float,
+        Field(
+            ...,
+            strict=True,
+            gt=0,
+            description="Maximum velocity in knots of the storm.",
+        ),
+    ]
+    clat: Annotated[
+        float,
+        Field(
+            ..., strict=True, ge=-90, le=90, description="Center latitude of the storm."
+        ),
+    ]
+    clon: Annotated[
+        float,
+        Field(
+            ...,
+            strict=True,
+            ge=-180,
+            le=180,
+            description="Center longitude of the storm.",
+        ),
+    ]
+    synoptic_time: datetime = Field(..., description="Synoptic time of the storm.")
+    aid_type: str = Field(..., description="Tropical cyclone forecast aid category.")
+    storm_num: Annotated[
+        int,
+        Field(
+            ...,
+            strict=True,
+            gt=0,
+            description=(
+                "Two digit storm number in sequential order from the start of the "
+                "hurricane season."
+            ),
+        ),
+    ]
+    storm_name: str = Field(description="Name of the storm.")
+    storm_basin: str = Field(
+        ...,
+        description=(
+            "Two character representation of the basin in which the storm originated "
+            "in."
+        ),
+    )
+    storm_year: Annotated[
+        int,
+        Field(
+            ...,
+            strict=True,
+            gt=1900,
+            description="Year in which the storm originated, after 1900.",
+        ),
+    ]
+    deck_line: str = Field(
+        ...,
+        description=(
+            "Deck data file used within the weather forecasting system for this storm."
+        ),
+    )
+    source_file: str = Field(..., description="File in which the storm data came from.")
+    final_storm_name: str = Field(
+        ...,
+        description="Final name of the storm used within the track file.",
+    )
+
+
+class VolcanoMetadata(FrozenModel):
```

Comment:
Will we be able to define sector models in plug-in repositories?

Reply:
Not with the way we've set this up currently, no. We could consider how to handle that, but there is no way to do this now.